common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -503,18 +515,34 @@ def dispatch_to_register():
         """Redirects to the registration page."""
         return redirect(AUTH_DISPATCH_URLS[AUTH_ENTRY_REGISTER])
 
-    def should_force_account_creation():
-        """ For some third party providers, we auto-create user accounts """
-        current_provider = provider.Registry.get_from_pipeline({'backend': backend.name, 'kwargs': kwargs})
+    def get_provider():
+        """
+        Get's third-party provider for request
+        """
+        return provider.Registry.get_from_pipeline({'backend': backend.name, 'kwargs': kwargs})
+
+    def should_autoprovision_account():
+        """ For some third party providers we trust the provider so much that we automatically provision the account """
+        current_provider = get_provider()
+        return current_provider and current_provider.autoprovision_account
+
+    def autosubmit_registration_form():
+        """ For some third party providers, we auto-submit registration forms """
+        current_provider = get_provider()
         return current_provider and current_provider.skip_email_verification
```

Comment:
Hmm, pretty sure there is a minor bug in the old code here. The check above should be for `skip_registration_form`, not `skip_email_verification`. What this is doing is: If the user is unrecognized and we are supposed to auto-submit the registration form, then redirect the user to the register form (which [will auto-submit](https://github.com/edx/edx-platform/blob/dbe74ea340deb96a8bad692233a816601e930c3c/lms/djangoapps/student_account/views.py#L214)) rather than the login form. In other words, when auto-submitting the form, we never want to show users the message that says "You are now signed in to provider X but not yet linked to an edX account".

If you want, this can be fixed in a separate PR though, probably against upstream.